src/compiler/compile/Component.ts

```diff
@@ -142,8 +143,20 @@ export default class Component {
 					.replace(process.cwd(), '')
 					.replace(regex_leading_directory_separator, '')
 				: compile_options.filename);
-		this.locate = getLocator(this.source, { offsetLine: 1 });
-
+		// line numbers in stack trace frames are 1-based. source maps are 0-based
+		const locator = getLocator(this.source, { offsetLine: 1 });
+		// @ts-ignore - fix the type of CompileOptions.sourcemap
+		const tracer = compile_options.sourcemap ? new TraceMap(compile_options.sourcemap) : undefined;
+		this.locate = (c) => {
+			let location: { line: number, column: number } = locator(c);
+			if (tracer) {
+				location = originalPositionFor(tracer, location);
+				// originalPositionFor returns 0-based lines?
+				// https://github.com/jridgewell/trace-mapping/issues/27
+				location.line += 1;
```

Comment:
judging from some other code in language tools and the fact that `tracemapping` says it is a drop-in replacement for `source-map` which has [line-1-based return values](https://github.com/mozilla/source-map#sourcemapconsumerprototypeoriginalpositionforgeneratedposition), this should not be done